### PR TITLE
feat(catalog): CAP-166 Cost Management - Supplier Item Cost Tiers

### PR DESCRIPTION
**Pull request by louisburroughs**
# CAP-166: Cost Management (Acquisition & Cost Models)

Implements backend services for supplier-item cost tier management as specified in [durion#166](https://github.com/louisburroughs/durion/issues/166).

## Backend Story
Closes [#195](https://github.com/louisburroughs/durion-positivity-backend/issues/195)

### New Features
- ✅ Supplier-item cost tier CRUD endpoints
- ✅ Cost tier management with effective dates
- ✅ Optimistic locking for concurrent updates
- ✅ Comprehensive validation and error handling
- ✅ Event emissions for audit trails

### Implementation Details

**Endpoints Added:**
- `GET /v1/products/suppliers/{supplierId}/items/{itemId}/costs` - List cost tiers
- `POST /v1/products/suppliers/{supplierId}/items/{itemId}/costs` - Create cost tier
- `PUT /v1/products/suppliers/{supplierId}/items/{itemId}/costs/{costId}` - Update cost tier
- `DELETE /v1/products/suppliers/{supplierId}/items/{itemId}/costs/{costId}` - Delete cost tier

**New Components:**
- `SupplierItemCostService` - Business logic for cost tier operations
- `SupplierItemCostRepository` - Data access layer
- `SupplierItemCostEntity` - JPA entity with optimistic locking
- `CostTierEntity` - Embedded cost tier value object
- Event type registrations in `CatalogEventTypes`

**Testing:**
- ✅ 7 module tests passing
- ✅ 24 contract behavioral tests passing
- ✅ Validation coverage for edge cases
- ✅ Concurrency conflict handling tested

## Documentation
- Contract guide: [domains/product/.business-rules/BACKEND_CONTRACT_GUIDE.md](https://github.com/louisburroughs/durion/blob/master/domains/product/.business-rules/BACKEND_CONTRACT_GUIDE.md)
- Implementation details: [CAP-166-backend-implementation.md](https://github.com/louisburroughs/durion/blob/master/docs/capabilities/CAP-166/CAP-166-backend-implementation.md)

## Verification

```bash
# Run module tests
./mvnw -pl pos-catalog test

# Run contract behavioral tests
./mvnw -pl pos-catalog -Dtest=ContractBehaviorIT test
```

## Architecture Compliance
- ✅ Follows internal package structure (service layer exposed, internals encapsulated)
- ✅ Uses `@NonNull` for null safety
- ✅ Implements `@EmitEvent` for audit logging
- ✅ Event type registration on startup
- ✅ Optimistic locking via `@Version`
- ✅ Validation with meaningful error messages

## Related Work
- Parent Capability: [durion#166](https://github.com/louisburroughs/durion/issues/166)
- Contract updates: Committed to [durion@cd58db2](https://github.com/louisburroughs/durion/commit/cd58db2)
- Contract PR: https://github.com/louisburroughs/durion/pull/302
- Frontend story: [durion-moqui-frontend#260](https://github.com/louisburroughs/durion-moqui-frontend/issues/260) (pending)

## Merge Requirements
- ✅ All tests passing
- ✅ Contract guide updated
- ✅ Implementation documented
- ⏳ Code review approval
- ⏳ CI/CD checks green

### CODE REVIEW

File: pos-catalog/src/main/java/com/positivity/catalog/service/SupplierItemCostService.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        if (baseCost != null && baseCost.signum() < 0) {
+            throw new CatalogValidationException("INVALID_DATA: baseCost cannot be negative.");
+        }
+    }
+
+    private void validateTierStructure(List<CostTierDto> tiers) {
+        if (tiers == null || tiers.isEmpty()) {
+            return;
+        }
+
```
[the remark this answers is not in the document]
Comment (louisburroughs) [reply pos-catalog/src/main/java/com/positivity/catalog/service/SupplierItemCostService.java]: Corrected

File: pos-catalog/src/main/java/com/positivity/catalog/service/SupplierItemCostService.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            if (minQuantity == null || minQuantity < 1) {
+                throw new CatalogValidationException("INVALID_DATA: min_quantity must be >= 1.");
+            }
+            if (unitCost == null || unitCost.signum() <= 0) {
+                throw new CatalogValidationException("INVALID_DATA: unit_cost must be a positive value.");
+            }
+            if (maxQuantity != null && minQuantity > maxQuantity) {
+                throw new CatalogValidationException("INVALID_TIER_STRUCTURE: min_quantity must be less than or equal to max_quantity.");
+            }
```
[the remark this answers is not in the document]
Comment (louisburroughs) [reply pos-catalog/src/main/java/com/positivity/catalog/service/SupplierItemCostService.java]: Corrected

File: pos-catalog/src/main/java/com/positivity/catalog/internal/dto/SupplierItemCostUpdateRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+@Data
+@Schema(description = "Request payload for updating supplier-item cost tiers")
+public class SupplierItemCostUpdateRequestDto {
+
+    @Schema(description = "ISO 4217 currency code", example = "USD", requiredMode = Schema.RequiredMode.REQUIRED)
+    private String currencyCode;
+
+    @Schema(description = "Optional fallback base cost when no tiers apply", example = "6.25")
+    private BigDecimal baseCost;
+
```
[the remark this answers is not in the document]
Comment (louisburroughs) [reply pos-catalog/src/main/java/com/positivity/catalog/internal/dto/SupplierItemCostUpdateRequestDto.java]: Comment removed version
